app/mailers/user_mailer.rb

```diff
@@ -12,14 +12,28 @@ class UserMailer < Devise::Mailer
 
   before_action :set_instance
 
+  before_deliver :verify_active_resource, only: %i(
+    backup_ready
+    confirmation_instructions
+    email_changed
+    password_change
+    reset_password_instructions
+    two_factor_disabled
+    two_factor_enabled
+    two_factor_recovery_codes_changed
+    webauthn_credential_added
+    webauthn_credential_deleted
+    webauthn_disabled
+    webauthn_enabled
+    welcome
+  )
```

Comment:
Maybe rather
```suggestion
  before_deliver :verify_active_resource, except: %i(
    warning
    appeal_approved
    appeal_rejected
    suspicious_sign_in
    failed_2fa
  )
```

Reply:
Makes sense, but while we're here and before committing that change ... SHOULD any of those have this same check, or do we want to send those ones out regardless of the check? (wraps account memorial)

Reply:
This is a good question. These checks were added in <https://github.com/mastodon/mastodon/pull/14726> without a rationale. On the one hand, most of the events behind the mails should not be happening at all if the user is marked as memorialized. On the other hand, it may make sense for the user or their close ones to receive the mails.

Reply:
Yeah, and prior to that the original disabled check was added in https://github.com/mastodon/mastodon/commit/1032f3994fdbd61c2f517057261ddc3559199b6b#diff-55254045ba774e4c360d7bbfde054665072d2ab49abc3f21add15a37bba65a1c - at the time it was added to all emails in the mailer.

What I'm not sure of is whether omission of the check in some emails is intentional and they should go out anyway, or if the check was just missed when new mails were added.

Reply:
Updated to add this check to all mailers here, on assumption that they should all have had it and missing were oversight.